components/apimgt/org.wso2.carbon.apimgt.gateway/src/main/java/org/wso2/carbon/apimgt/gateway/listeners/GatewayJMSMessageListener.java

```diff
@@ -55,6 +58,7 @@
     private InMemoryAPIDeployer inMemoryApiDeployer = new InMemoryAPIDeployer();
     GatewayArtifactSynchronizerProperties gatewayArtifactSynchronizerProperties = ServiceReferenceHolder
             .getInstance().getAPIManagerConfiguration().getGatewayArtifactSynchronizerProperties();
+    ScheduledExecutorService scheduler = Executors.newScheduledThreadPool(10);
```

Comment:
Make this a constant.

Reply:
https://github.com/wso2/carbon-apimgt/pull/9016/commits/5e8ba5419c5fc20d7981d5a593400c8ceecf9d2b